# Kya Allen - KH06059

In [62]:
import pandas as pd
import math

In [106]:
df = pd.read_csv('5SampleData.csv')
df.describe()

A          B          C          D          E
count  15.000000  15.000000  15.000000  15.000000  15.000000
mean    1.801179   2.884182   4.435853   4.399258   5.334068
std     1.641574   2.147900   6.011582   4.544478   6.090725
min     0.169492   0.080756   0.053384   0.021413   0.211548
25%     0.726072   1.241924   0.576944   1.652986   1.130329
50%     1.167329   2.396752   1.137213   2.209060   3.335929
75%     2.927610   4.271409   5.553287   6.568086   7.212631
max     4.530944   6.821814  20.604758  16.541439  24.185195

In [105]:
Xbars = []
S2s = []
N = []
#Make a vector of sample means
for i in df.mean():
    Xbars.append(i)
#Make a Vector of Sample Varainces
for i in df.std():
    S2s.append(i**2.0)
#Make a Vector of Sample Sizes
for i in df.count():
    N.append(i)

In [58]:
#Get a pooled mean
def PooledMean(X, V, N):
    numerator = 0
    Denom = 0
    count = len(X)
    NumerList = []
    DenomList = []
    for i in range(count):
        NumerList.append((N[i] * X[i]) / V[i])
        DenomList.append(N[i] / V[i])
    for i in NumerList:
        numerator = numerator + i
    for i in DenomList:
        Denom = Denom + i
        
    theta = numerator / Denom    
    return theta
#Get a Pooled Variance
def PooledVariance(V, N):
    Denom = 0
    count= len(V)
    DenomList = []
    for i in range(count):
        DenomList.append(N[i] / V[i])
    for i in DenomList:
        Denom = Denom + i
    VarPool = 1 / Denom
    return VarPool

In [104]:
PM = PooledMean(Xbars, S2s, N)
PV = PooledVariance(S2s, N)

In [69]:
#Hypethesis Test
def Ztest(NullMean, ObservedMean, Variance):
    Zscore = (ObservedMean - NullMean) / math.sqrt(Variance)
    return Zscore

In [70]:
Ztest(4, PM, PV)

-4.615083434442804

##The Zscore, -4.62 is Less than the 95% rejection threshold 1.65. We fail to reject the null hypothesis

In [72]:
##Part B:
#Get proportion bellow 3

In [82]:
proportions = []
for i in df.columns:
    Successes = 0
    for j in df[i]:
        if j <= 3:
            Successes = Successes + 1
    proportions.append(Successes / 15)
proportions

[0.7333333333333333, 0.6, 0.6, 0.6, 0.4666666666666667]

In [100]:
PropVariances = []
for i in proportions:
    PropVariances.append((i * (1 - i)) / 15)
PropVariances

[0.01303703703703704, 0.016, 0.016, 0.016, 0.016592592592592593]

In [107]:
##use the pooled mean function, while passing proportions instead of means
PP = PooledMean(proportions, PropVariances, N)
PP

0.6067542213883678

In [102]:
#use the pooled variance function, while passing Proportion variances
PPV = PooledVariance(PropVariances, N)
PPV

0.00020546174692516158

In [108]:
Ztest(0.30, PP, PPV)

21.400556150343782

In [ ]:
##The Z score, 21.40 is greater than the the critical level 1.28 for a 90% threshold. We reject the Null Hypothesis.